### PR TITLE
Add NTP SI creative total metric

### PR DESCRIPTION
**Pull request by DJAndries**
Resolves brave/brave-browser#28012

## Submitter Checklist:

- [ ] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [ ] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [ ] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [ ] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [ ] Squashed any review feedback or "fixup" commits before merge, so that history is a record of what happened in the repo, not your PR
- [ ] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally:
  * `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests` [wiki](https://github.com/brave/brave-browser/wiki/Tests)
  * `npm run lint`, `npm run presubmit` [wiki](https://github.com/brave/brave-browser/wiki/Presubmit-checks), `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

0. Start with fresh profile, ensure ads is disabled, ensure no metrics with the prefix creativeInstanceId exist in the local state page.
1. Open new tabs until an ad is shown.
2. Ensure that metric is not sent immediately.
3. Move system time forward by one day, launch browser. Verify that the metric is sent (by using a mitm proxy), with the URL https://p3a-creative.brave.com set. Ensure the metric no longer exists on the local state page. Ensure the metric has a value of 1.
4. Close browser, advance the time by one day, open browser, ensure metric is sent with a value of 0.
5. Enable ads, close browser, advance the time by one day, open browser, ensure that the metric is not sent.

### CODE REVIEW

**Review by iefremov — APPROVED** [review]

File: browser/ntp_background/ntp_p3a_helper_impl_unittest.cc [hunk 1]
```diff
@@ -44,7 +47,7 @@ class NTPP3AHelperImplTest : public testing::Test {
         new brave::BraveP3AService(&local_state_, "release", "2049-01-01"));
 
     ntp_p3a_helper_.reset(
-        new NTPP3AHelperImpl(&local_state_, p3a_service_.get()));
+        new NTPP3AHelperImpl(&local_state_, p3a_service_.get(), &ads_service_));
```
Comment (tmancey) [inline browser/ntp_background/ntp_p3a_helper_impl_unittest.cc:50]: nit: use `std::make_unique`

File: browser/ntp_background/ntp_p3a_helper_impl.h [hunk 2]
```diff
@@ -57,6 +62,7 @@ class NTPP3AHelperImpl : public NTPP3AHelper {
 
   raw_ptr<PrefService> local_state_;
   raw_ptr<brave::BraveP3AService> p3a_service_;
+  raw_ptr<brave_ads::AdsService> ads_service_;
```
Comment (tmancey) [inline browser/ntp_background/ntp_p3a_helper_impl.h:65]: nit: can be `const`

**Review by tmancey — APPROVED** [review]
LGTM (see nits)

**Comment by GeetaSarvadnya** [comment]
Verification IN PROGRESS on

```
Brave | 1.50.56 Chromium: 110.0.5481.104 (Official Build) nightly (64-bit)
-- | --
Revision | 46de4a7f41979e829b430bc1ee30ef483aa227ac-refs/branch-heads/5481_77@{#19}
OS | Windows 10 Version 21H2 (Build 19044.2604)

```
- Verified the test plan from https://github.com/brave/brave-core/pull/16836#issue-1555540945

**Comment by GeetaSarvadnya** [comment]
@DJAndries Can we review the steps 3 and step 5 mentioned above? I am seeing unexpected results in these steps. Looks like the functionality is not working as per the test plan

**Comment by GeetaSarvadnya** [comment]
Verification **PASSED** on

```
Brave | 1.50.61 Chromium: 111.0.5563.33 (Official Build) nightly (64-bit)
-- | --
Revision | 19bd6d0186b7912604e18191826dab9d1df00e2e-refs/branch-heads/5563@{#557}
OS | Windows 10 Version 21H2 (Build 19044.2604)

```
- Verified the test plan from https://github.com/brave/brave-core/pull/16836#issue-1555540945

**Step 0**: 

 - Clean profile 1.50.61
 - Ensured Ads is disabled
 - Ensured no metrics with the prefix creativeInstanceId exist in the local state page under P3A logs in brave://local-state file
 
**Step 1:**
 - Open NTP until NTP SI is shown
 - Ensured NTP SI is shown in 2nd tab
 
image

**Step 2:**
 - Ensured that the metrics value `1` is not sent immediately over the network via https://p3a-creative.brave.com/
image

**Step 3:**
  - Closed the browser and set the sys date ahead by one day
  - Relaunch the browser
  - Ensured that the metrics value `1` is sent over the network via https://p3a-creative.brave.com/
image (3)


**Step 4:**
 - Closed the browser and set the sys date ahead by one day
 - Relaunch the browser
 - Ensured that the metrics value `0` is sent over the network via https://p3a-creative.brave.com/
 
image (4)

**Step 5:**
 - Enabled ads 
 - Closed the browser and set the sys date ahead by one day
 - Relaunch the browser
 - Ensured that the metrics is not sent over the network via https://p3a-creative.brave.com/
 
Example | Example
----------|--------
image (5) | image (6)

**Pull request merged**